libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/guidance/restarea/RestAreaProcessor.kt

```diff
@@ -116,18 +111,31 @@ internal object RestAreaProcessor {
         )
     }
 
-    private fun RouteProgress.getFirstUpcomingRoadObject(
-        @RoadObjectType.Type type: Int
-    ): UpcomingRoadObject? {
-        return this.upcomingRoadObjects.firstOrNull {
-            if (it.roadObject.objectType == type) {
+    private fun RouteProgress.getUpcomingRestStopOnCurrentStep(): RestStop? {
+        val upcomingRestStop = getFirstUpcomingRestStop() ?: return null
+        val upcomingRestStopLocation = upcomingRestStop.location.shape as? Point ?: return null
+
+        val currentStep = currentLegProgress?.currentStepProgress?.step
+        val isUpcomingRestStopAtCurrentStep =
+            !currentStep?.getRestStopsMatchingLocation(upcomingRestStopLocation).isNullOrEmpty()
```

Comment:
```suggestion
        val currentStepProgress = currentLegProgress?.currentStepProgress ?: return null
        val isUpcomingRestStopAtCurrentStep = upcomingRestStop.distanceToStart - currentStepProgress.distanceRemaining <= 0
```
This still looks to me like an approach worth considering to avoid point precision issues and just speed up the logic.

Reply:
Great idea. Thanks.

Reply:
Done. Please update version.